anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -770,202 +848,87 @@ def _run_grypedb_sync_task(self, checksum: str, grype_db_data: bytes) -> None:
         :type group_download_result: GroupDownloadResult
-        :param full_flush: whether or not to flush out old records before syncing
-        :type full_flush: bool, defaults to False
+        :param group_obj: metadata record for this feed group
+        :type group_obj: FeedGroupMetadata
         :param local_repo: LocalFeedDataRepo (disk cache for download results)
         :type local_repo: Optional[LocalFeedDataRepo], defaults to None
-        :param operation_id: operation id (UUID4 in hexadecimal)
-        :type operation_id: Optional[str], defaults to None
-        :return:
+        :return: int
+        :rtype: total number of records updated
         """
-        result = build_group_sync_result()
-        result["group"] = group_download_result.group
-
-        db = get_session()
-        group_db_obj = None
-        if self.metadata:
-            db.refresh(self.metadata)
-            group_db_obj = self.group_by_name(group_download_result.group)
-
-        if not group_db_obj:
-            logger.error(
-                log_msg_ctx(
-                    operation_id,
-                    group_download_result.feed,
-                    group_download_result.group,
-                    "Skipping sync for feed group {}, not found in db, record should have been synced already",
-                )
-            )
-            return result
-
-        download_started = group_download_result.started.replace(
-            tzinfo=datetime.timezone.utc
-        )
-        sync_started = time.time()
-        total_updated_count = 0
-        try:
-            if full_flush:
-                logger.info(
-                    log_msg_ctx(
-                        operation_id,
-                        group_download_result.feed,
-                        group_download_result.group,
-                        "Performing data flush prior to sync as requested",
-                    )
-                )
-                self._flush_group(
-                    group_db_obj,
-                    operation_id=operation_id,
-                )
-
-            # Iterate thru the records and commit
-
-            logger.info(
-                log_msg_ctx(
-                    operation_id,
-                    group_download_result.feed,
-                    group_download_result.group,
-                    "Syncing {} total update records into db in sets of {}".format(
-                        group_download_result.total_records, self.RECORDS_PER_CHUNK
-                    ),
+        total_records_updated = 0
+        for record in local_repo.read_files(
+            group_download_result.feed, group_download_result.group
+        ):
+            # If we go through two files, then that means the feed service provided two GrypeDB files.
+            # This is an unexpected condition.
+            if total_records_updated >= 1:
+                raise UnexpectedRawGrypeDBFile()
```

Comment:
If this is getting thrown in multiple places, it might be easier to debug if you add a message with some context to this.

Reply:
only gets thrown here